src/Application.php

```diff
@@ -45,6 +45,10 @@ public function __construct($name = 'UNKNOWN', $version = 'UNKNOWN')
         $this->expressionParser = new ExpressionParser();
         $this->invoker = new Invoker();
 
+        if ($this->invoker->getParameterResolver() instanceof ResolverChain) {
```

Comment:
Yes I think it would be better to pass all the resolvers explicitly, that would be clearer.